campbot/checkers.py

```diff
@@ -486,3 +487,24 @@ def __init__(self):
 
     def test_contribution(self, old_doc, new_doc):
         return old_doc.quality == new_doc.quality
+
+class TitleLevel(object):
+    def __init__(self):
+        self.name = "Titre de niveau trop élevé"
+        self.fail_marker = emoji("/uploads/default/original/2X/0/0178043b1b70e669946f609571bd4b8f7d18e820.png",
+                                 self.name)
+        self.success_marker = ""
+
+    def __call__(self, contrib, old_version, new_version):
+        if old_version is None or new_version is None:
+            return True, True
+
+        old_doc = old_version.document
+        new_doc = new_version.document
+
+        if "redirects_to" in old_doc or "redirects_to" in new_doc:
+            return True, True
+
+        title =  not new_doc.search( r"(^|\n)##[^#]") == None
+        subtitle = not new_doc.search(r"(^|\n)###*" == None
+        return True, not subtitle or title
```

Comment:
Here is the subtile part. With this result, you will say taht the old version is always ok, and the new one is good or not.

It means that even if the contributor did not changed exactly this part, the contribution can be reported if the error was previously here. 

As this process is meant to prevent quality to degrade, but not check existing errors, it can lead to an overflow of contribution reported.

To fix this, we need to encapsulate the checks into a function, and call it on both version: 

Suggestion for the last 3 lines : 

  ```suggestion
        def check(doc):
            has_level2 =  doc.search( r"(^|\n)##[^#]") is not None
            has_level3_or_more = doc.search(r"(^|\n)###*" is not None
            
            return not has_level2 and has_level3_or_more
        
        return check(old_doc), check(new_doc)
```